File: python/linear_search.py
import random

def search(lst: list[int], val: int) -> None:
    for i, j in enumerate(lst):
        if lst[i] == val: print(f"Value {j} found at index {i}") 

def gen_random_list(l: int, min: int, max: int): 
    lst = []
    for _ in range(l):
        lst.append(random.randint(min, max))

    find_val = lst[random.randint(0, (len(lst) - 1))]
    return lst, find_val


if __name__ == "__main__":
    l, find = gen_random_list(10, 0, 100)
    print(l)
    search(l, find)

File: javascript/linear_search.js
function linear_search(arr, find_val) {
  for (let i = 0; i < arr.length; i++) {
    if (arr[i] == find_val) {
      return i
    }
  }
}

let genRandomList = (long) => {
  let arr = []; 
  for (let i = 0; i < long; i++) {
    let add = (Math.floor(Math.random() * 100 + 1))
    arr.push(add)
  }
  return arr 
}

let arr = genRandomList(10)
let find = arr[Math.floor(Math.random() * (arr.length + 1))]
console.log(find)
console.log(arr)
let search = linear_search(arr, find)

console.debug(search)
